mrp_subcontracting_purchase/models/purchase_order.py

```diff
@@ -0,0 +1,119 @@
+            )
+        return productions
+
+    def action_view_picking(self):
+        return self._get_action_view_picking(self.picking_ids)
+
+    def _get_action_view_picking(self, pickings):
+        """This function returns an action that display existing picking orders
+        of given purchase order ids. When only one found, show the picking immediately.
+        """
+        self.ensure_one()
+        result = self.env["ir.actions.actions"]._for_xml_id(
+            "stock.action_picking_tree_all"
+        )
+        # override the context to get rid of the default filtering on operation type
+        result["context"] = {
+            "default_partner_id": self.partner_id.id,
+            "default_origin": self.name,
+            "default_picking_type_id": self.picking_type_id.id,
+        }
+        # choose the view_mode accordingly
+        if not pickings or len(pickings) > 1:
+            result["domain"] = [("id", "in", pickings.ids)]
+        elif len(pickings) == 1:
+            res = self.env.ref("stock.view_picking_form", False)
+            form_view = [(res and res.id or False, "form")]
+            result["views"] = form_view + [
+                (state, view)
+                for state, view in result.get("views", [])
+                if view != "form"
+            ]
+            result["res_id"] = pickings.id
+        return result
+
+    def _get_destination_location(self):
+        """Returns destination location for subcontractor"""
+        self.ensure_one()
+        if not self.dest_address_id or self.sale_order_count:
+            return super()._get_destination_location()
+
+        mrp_production_ids = self._get_mrp_productions(
+            remove_archived_picking_types=False
+        )
+
+        if (
+            mrp_production_ids
+            and self.dest_address_id in mrp_production_ids.bom_id.subcontractor_ids
+        ):
+            return self.dest_address_id.property_stock_subcontractor.id
+
+        in_bom_products = False
+        not_in_bom_products = self.env["purchase.order.line"]
+        for order_line in self.order_line:
+            if any(
+                bom_line.bom_id.type == "subcontract"
+                and self.dest_address_id in bom_line.bom_id.subcontractor_ids
+                for bom_line in order_line.product_id.bom_line_ids.filtered(
+                    lambda line: line.company_id == self.company_id
+                )
+            ):
+                in_bom_products = True
+            elif not order_line.display_type:
+                not_in_bom_products |= order_line
+        if in_bom_products and not_in_bom_products:
+            raise UserError(
+                _(
+                    """It appears some components in this RFQ are not meant for subcontracting.
+                    Please create a separate order for these."""
+                )
+                + "\n\n"
+                + "\n".join(not_in_bom_products.mapped("name"))
```

Comment:
It's not good idea, you must use format like this for one variable `_("....... \n\n%s", value)`.